platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/ValueState.java

```diff
@@ -231,7 +231,9 @@ public void beforeMutate(Describable displayName) {
             } else if (warnOnUpgradedPropertyChanges) {
                 String shownDisplayName = displayName.getDisplayName();
                 DeprecationLogger.deprecateBehaviour("Changing property value of " + shownDisplayName + " at execution time.")
-                    .startingWithGradle9("changing property value of " + shownDisplayName + " at execution time is deprecated and will fail in Gradle 10")
+                    // this should only happen in Gradle 10, when Provider API migration will come to the mainline,
+                    // so forbidding it must wait until Gradle 11
+                    .startingWithGradle11("changing property value of " + shownDisplayName + " at execution time will become an error")
                     // TODO add documentation
                     .undocumented()
```

Comment:
Can we add some documentation for this?

Reply:
Not yet, but we should (once we master is prepared for Gradle 10).

Reported: https://github.com/gradle/gradle/issues/33687